dashboard/optional_utils.py

```diff
@@ -242,3 +249,51 @@ def _update_cache(task):
         return _wrapper(target_func)
     else:
         return _wrapper
+
+
+def init_ray_and_catch_exceptions(connect_to_serve: bool = False) -> Callable:
+    """Decorator to be used on methods that require being connected to Ray."""
+
+    def decorator_factory(f: Callable) -> Callable:
+        @functools.wraps(f)
+        async def decorator(self, *args, **kwargs):
+            try:
+                if not ray.is_initialized():
+                    try:
+                        if use_gcs_for_bootstrap():
+                            address = self._dashboard_head.gcs_address
+                            redis_pw = None
+                            logger.info(f"Connecting to ray with address={address}")
+                        else:
+                            ip, port = self._dashboard_head.redis_address
+                            redis_pw = self._dashboard_head.redis_password
+                            address = f"{ip}:{port}"
+                            logger.info(
+                                f"Connecting to ray with address={address}, "
+                                f"redis_pw={redis_pw}"
+                            )
+                        ray.init(
+                            address=address,
+                            namespace=RAY_INTERNAL_DASHBOARD_NAMESPACE,
+                            _redis_password=redis_pw,
+                        )
+                    except Exception as e:
+                        ray.shutdown()
+                        raise e from None
+
+                if connect_to_serve:
+                    # TODO(edoakes): this should probably run in the `serve`
+                    # namespace.
+                    serve.start(detached=True)
```

Comment:
Are the deployments deployed correctly without starting Serve in the dashboard's namespace?

Reply:
We need to fix this more generally though so things don't start in this wonky namespace